File: src/main/java/com/booknav/springbackend/utils/response/Response.java
package com.booknav.springbackend.utils.response;

public class Response {
    public static <T> ResponseModel<T> successRes(String message, T body) {
        return new ResponseModel<>("success", message, body);
    }

    public static <T> ResponseModel<T> errorRes(String message, T body) {
        if (body == null) {
            body = (T) "No data exists!";
        }
        return new ResponseModel<>("failure", message, body);
    }
}

File: ASP Core - Backend/Utils/Response/Response.cs
namespace aspcorebackend.Utils.Response {
    public class Response<T> {
        public string Status { get; set; } = null!;
        public string Message { get; set; } = null!;
        public T? Body { get; set; }

        public Response(string status, string message, T? body) {
            Status = status;
            Message = message;
            Body = body;
        }
    }
}
